packages/datadog-plugin-aws-sdk/src/index.js

```diff
@@ -29,18 +29,37 @@ function createWrapRequest (tracer, config) {
         tags
       })
 
-      const boundCb = typeof cb === 'function' ? tracer.scope().bind(cb, childOf) : cb
-
       this.on('complete', response => {
         if (!span) return
 
-        awsHelpers.addResponseTags(span, response, serviceName, config)
+        awsHelpers.addResponseTags(span, response, serviceName, config, tracer)
         awsHelpers.finish(span, response.error)
       })
 
       analyticsSampler.sample(span, config.analytics)
 
+      awsHelpers.requestInject(span, this, serviceName, tracer)
+
+      const request = this
+
       return tracer.scope().activate(span, () => {
+        let boundCb
+        if (typeof cb === 'function') {
+          boundCb = function wrappedCb (err, resp) {
```

Comment:
This logic should be pushed to the service helper.

Reply:
Hmm, alright I'll give that a shot.